pkg/core/blockchain.go

```diff
@@ -69,16 +69,16 @@ func NewBlockchain(s storage.Store, cfg config.ProtocolConfiguration) (*Blockcha
 		blockCache:    NewCache(),
 		verifyBlocks:  false,
 	}
-	go bc.run()
 
+	go bc.run()
```

Comment:
there's no possible to stop BlockChain?.. we must use ctx (context.Context) that passed to `NewBlockchain`...

and than `bc.run`: 

```go
func (bc *Blockchain) run(ctx context.Context) {
	persistTimer := time.NewTimer(persistInterval)
	for {
		select {
		case <-ctx.Done():
			return
		case op := <-bc.headersOp:
			op(bc.headerList)
			bc.headersOpDone <- struct{}{}
		case <-persistTimer.C:
			go bc.persist()
			persistTimer.Reset(persistInterval)
		}
	}
}
```

Reply:
Does this change need to be done in this here? Can we do later by opening a specific issue for this? I am afraid If I add a new argument in the NewBlockchain, I will need to touch other parts of the code which are not related to this PR.

Reply:
Yeah.. you right